hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -153,11 +156,14 @@ protected <T> ClosableIterator<T> deserializeRecords(HoodieReaderContext<T> read
   }
 
   private static class RecordIterator implements ClosableIterator<IndexedRecord> {
+
+    private final Boolean whichImplementation = false;
     private byte[] content;
     private final SizeAwareDataInputStream dis;
     private final GenericDatumReader<IndexedRecord> reader;
     private final ThreadLocal<BinaryDecoder> decoderCache = new ThreadLocal<>();
-
+    private Option<Schema> castSchema = Option.empty();
```

Comment:
maybe rename to `evolvedSchema`?

Reply:
Renamed to `promotedSchema` and updated some other method names to make it clear that this is for type promotion and not schema evolution as a whole